### PR TITLE
[ADD] add credit points to navbar

### PR DESCRIPTION
**Pull request by mpanarin**
this adds credit points to website navbar

### CODE REVIEW

File: website_sale_credit_point/i18n/fr.po [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+"Language-Team: \n"
+"MIME-Version: 1.0\n"
+"Content-Type: text/plain; charset=UTF-8\n"
+"Content-Transfer-Encoding: \n"
+"Plural-Forms: \n"
+
+#. module: website_sale_credit_point
+#: model:ir.ui.view,arch_db:website_sale_credit_point.navbar_points
+msgid "<span>Credit balance: </span>"
+msgstr "<span>Solde disponible: </span>"
```
Comment (asaunier) [inline website_sale_credit_point/i18n/fr.po]: Minor but in French colons always have a (non breaking) space in front of them (eg. "Solde disponible :")

File: website_sale_credit_point/templates/navbar_points.xml [hunk 2]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<odoo>
+
+    <template id="navbar_points" inherit_id="portal.frontend_layout">
+        <xpath expr="//ul[@id='top_menu']/*[1]" position="before">
+            <li t-if="user_id.partner_id">
+                <!-- i use empty a tag here to get the styles, used in top menu -->
```
Comment (asaunier) [inline website_sale_credit_point/templates/navbar_points.xml]: "empty a tag"? Perhaps you mean "an empty tag"?

File: website_sale_credit_point/templates/navbar_points.xml [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<odoo>
+
+    <template id="navbar_points" inherit_id="portal.frontend_layout">
+        <xpath expr="//ul[@id='top_menu']/*[1]" position="before">
```
Comment (simahawk) [inline website_sale_credit_point/templates/navbar_points.xml]: why this selector? I bet UL is going to contain LI elems

File: website_sale_credit_point/templates/navbar_points.xml (hunk 2 above)
Comment (mpanarin) [reply website_sale_credit_point/templates/navbar_points.xml]: No, that's not what I meant, but I will rephrase it to be more clear :D

**Review by simahawk — CHANGES_REQUESTED** [review]

File: website_sale_credit_point/templates/navbar_points.xml (hunk 3 above)
Comment (mpanarin) [reply website_sale_credit_point/templates/navbar_points.xml]: yes, and i want my element before any other LI element, so i used this selector instead of using `position="inside"`

Comment (simahawk) [reply website_sale_credit_point/templates/navbar_points.xml]: then why a wildcard? use `//ul[@id='top_menu']/li[1]`.
 `*` must be used only when you have no other chance

**Review by TDu — APPROVED** [review]

File: website_sale_credit_point/templates/navbar_points.xml (hunk 3 above)
Comment (mpanarin) [reply website_sale_credit_point/templates/navbar_points.xml]: this doesn't work. As I need it as a far left element. And the buttons like `Home`, etc. are not specified in the templates. If I specify it as `//ul[@id='top_menu']/li[1]` it will be before custom added buttons, like `Cart` but not the base one.

Comment (yvaucher) [reply website_sale_credit_point/templates/navbar_points.xml]: @mpanarin I don't see how a dependency issue with `Cart` is solved that way. It should be the exact same result then no? I don't get why * would play differently, is it evaluated later?

Not sure if possible with templates, but could we not play with priority set to the lowest to ensure this one is added ?

Otherwise, maybe search for `<li>` with `<a href='/'>` I think we can assume that button `Home` will always be there.

File: website_sale_credit_point/templates/navbar_points.xml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+<?xml version="1.0"?>
+<odoo>
+
+    <template id="navbar_points" inherit_id="website.layout">
+        <xpath expr="//t[@t-foreach='website.menu_id.child_id']" position="before">
```
Comment (simahawk) [inline website_sale_credit_point/templates/navbar_points.xml:5]: :+1:

**Review by simahawk — APPROVED** [review]
